sklearn/tests/test_dummy.py

```diff
@@ -101,6 +100,23 @@ def test_most_frequent_and_prior_strategy():
                                       clf.class_prior_.reshape((1, -1)) > 0.5)
 
 
+def test_most_frequent_and_prior_strategy_with_pandas_dataframe():
```

Comment:
Good stuff this is a non-regression test indeed since this fails in master. I think you should rename it though since it does not use a pandas dataframe at all.